### PR TITLE
Homework Catch2

### PR DESCRIPTION
**Pull request by arafalski**
Authors: @majkel84, @Vicx95, @Nessy90, @mateusz-talma, @Kasperice

### CODE REVIEW

**Review by WBigaj — COMMENTED** [review]
I have checked your code and it looks good :)

File: bowling/testingScenarios [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+            THEN ("number of rounds is too small") {
+                """REQUIRE EXCEPTION"""
+            }
+        }
+    }
+}
+
+SCENARIO("check score vector size with no spare or strike at the end"){
+    GIVEN("sccore vector of pairs") {
+        std::vector<std::pair<int,int>> v {};
```
Comment (ziobron) [inline bowling/testingScenarios:341]: why vector of pair here? It was just vector of ints above

File: bowling/testingScenarios [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+    GIVEN("empty vecotr of pairs") {
+        std::vector<pair<int,int>> v{};
+        WHEN ("check all elements after game is played") {
+            const size_t rounds = 10;
+            playGame(rounds);
+            THEN ("all elements of vector should be positive") {
+                bool arePositive = TRUE;
+                for (const auto& it: v) {
+                    if (*it.first < 0 || *it.second < 0) {
+                        arePositive = FALSE;
```
Comment (ziobron) [inline bowling/testingScenarios:382]: you can add `break;` after `arePositive = false` or use std::all_of / std::any_of / std::none_of

File: bowling/testingScenarios [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+}
+                
+                
+SCENARIO("each turn have maximum two tries"){
+    GIVEN("triesCounter and turnCounter") {
+        int turnCounter = 1;
+        int triesCounter = 0;
+        int maxTries = 2;
+        playGame(turnCounter);
+        WHEN ("triesCounter == maxTries") {
```
Comment (ziobron) [inline bowling/testingScenarios:411]: how triesCounter will be incremented? It is not passed into playGame function.

File: bowling/testingScenarios [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+        int turnCounter = 1;
+        WHEN ("ball knocks down all pins") {
+                turnCounter++;
+            THEN ("Increment turnCounter") {
+                int expectedValue = 2;
+                REQUIRE_THAT(turnCounter, Equals(expectedValue));
+             }
+        }
+    }
+}
```
Comment (ziobron) [inline bowling/testingScenarios:431]: not a test scenario. It's rather implementation detail.

File: bowling/testingScenarios [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+        std::vector<std::pair<int, int>> score { {0,10}, {8, 0} }
+        WHEN ("second ball knocks down all pins") {
+            THEN ("Add to totalScore score of next ball and totalScore should equal to expectedValue") {
+                int totalScore = countScore(score);
+                int expectedValue = 26;
+                REQUIRE_THAT(totalScore, Equals(expectedValue));
+            }
+        }
+    }
+}
```
Comment (ziobron) [inline bowling/testingScenarios:457]: Two above cases were already covered

File: bowling/testingScenarios [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+                }
+                REQUIRE(arePositive);  
+            }
+        }
+    }
+}
+
+SCENARIO("game have ten turns"){
+    GIVEN("counter of turns") {
+        int turnCounter = 0;
```
Comment (ziobron) [inline bowling/testingScenarios:393]: turnCounter will be an internal detail of playGame function. It is not a test case.

File: bowling/testingScenarios [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+    GIVEN("game with too less rounds") {
+        std::vector<int> v{0,0, 0,0};
+        WHEN ("count score after game") {
+            int score = countScore(v);
+            THEN ("number of rounds is too small") {
+                """REQUIRE EXCEPTION"""
+            }
+        }
+    }
+}
```
Comment (ziobron) [inline bowling/testingScenarios:337]: all above scenarios look good. I could count 28 of them

File: bowling/testingScenarios [hunk 8]
```diff
@@ -0,0 +1,458 @@
...
+
+SCENARIO("check all vector elements are digit"){
+    GIVEN("empty vector of pairs") {
+        std::vector<pair<int,int>> v{};
+        WHEN ("check all elements after game is played") {
+            const size_t rounds = 10;
+            playGame(rounds);
+            THEN ("all elements of score vector should be digit") {
+                auto& it = find_if_not(v.begin(), v.end(), isdigit);
```
Comment (ziobron) [inline bowling/testingScenarios:365]: type int doesn't allow anything other that numbers to be stored. std::isdigit works on char.

File: bowling/testingScenarios [hunk 9]
```diff
@@ -0,0 +1,458 @@
...
+                REQUIRE(it == v.end());  
+            }
+        }
+    }
+}
+
+SCENARIO("check all vector elements are positive"){
```
Comment (ziobron) [inline bowling/testingScenarios:372]: I can also count in this one test case.

**Review by ziobron — APPROVED** [review]
29 cases 👍

**Comment by ziobron** [comment]
I have a remark that it would be useful to see the connections between played balls and final score. You could use some constexpr to show where they are added in a final score.

**Comment by ziobron** [comment]
{
  "requirement_ids": [3883, 1636],
  "points": [29, 3],
  "users": ["majkel84", "Vicx95", "Nessy90", "mateusz-talma", "Kasperice"]
}

**Comment by ziobron** [comment]
{
"requirement_ids": [3883, 1636],
"points": [29, 3],
"users": ["anraf1001"]
}

**Pull request closed without merging**